KsApi/models/ValidateCheckoutEnvelope.swift

```diff
@@ -14,6 +14,11 @@ extension ValidateCheckoutEnvelope {
     guard let envelope = ValidateCheckoutEnvelope.validateCheckoutEnvelope(from: data) else {
       return SignalProducer(error: .couldNotParseJSON)
     }
+
+    if envelope.valid == false {
```

Comment:
This doesn't make sense to me. If we're always going to say that `valid == false` means we create an error envelope instead, why would we have a `valid` field at all on the `ValidateCheckoutEnvelope`? What do you think of either a) using the current `valid` field in the postCampaignCheckoutVM to decide if we should show an error banner or not or b) getting rid of the `valid` field entirely and instead create the errorEnvelope immediately once we see the checkout isn't valid?

Reply:
🤔 Yea we could remove `valid` from `ValidateCheckoutEnvelope` and instead of 
`if envelope.valid == false {...}` we could just check the `checkout` object directly like this 

`guard data.checkout?.isValidForOnSessionCheckout.valid == true else {...}`

Is that what you mean?

Reply:
I do think that it's useful to differentiate between .couldNotParseJSON and .validateCheckoutError ErrorEnvelopes though.

I also considered removing `messages` in the same way because the extra `validateCheckoutEnvelope()` seems excessive and not super useful for this case.

Reply:
I went ahead and[ removed `valid` from the envelope](https://github.com/kickstarter/ios-oss/pull/2053#:~:text=8ba4db2-,remove%20redundant%20valid%20property%20from%20ValidateCheckoutEnvelope,-c70d444) so that we just have the one!

As for updating the UX so that users who attempt to pledge with an invalid payment method, I already have a separate [ticket](https://kickstarter.atlassian.net/browse/MBL-1422) to address this 😄 . Bessie should be adding an `errorTypes` array to this mutation that I think we can expose and then use to do that.